Create disable-alerts-health-management-api.md (#112)
netdata/blog: blog/disable-alerts-health-management-api.md

---
slug: Mute Alerts 
title: "How to mute alerts during maintenance windows or scheduled backups?"
description: "Disable or Mute Alerts during maintenance windows"
tags: [netdata,alerts,silence,disable,mute,maintenance window]
keywords: [netdata,alerts,silence,disable,mute,maintenance window]
authors: satya
---

For all SREs, it is absolutely crucial to filter out expected events during maintenance windows and quickly pinpoint critical issues in your infrastructure. Every minute is crucial while dealing with troubleshooting issues and any distractions that may hijack the troubleshooting process should be subdued.
The health management APIs in Netdata allows teams to eliminate unnecessary alerting during scheduled maintenance, testing, auto scaling events, and instance reboots.
 
## Health Management API

> Netdata v1.12 and beyond provides a command API to control health checks and notifications at runtime. The feature is especially useful for maintenance periods, during which you receive meaningless alerts. From Netdata v1.16.0 and beyond, the configuration controlled via the API commands is [persisted across Netdata restarts](https://learn.netdata.cloud/docs/agent/web/api/health#persistence).

Reviewer: I think, this far ahead of the version, we can say (for the blog at least), just "Netdata provides a command API..."

Same with the next mention. Let's just say it exists without version specificity. 

You okay with that, @cakrit?

Reviewer 2: Yes, that version is ancient.

Author: This is fixed.

---
slug: Mute Alerts 
title: "How to mute alerts during maintenance windows or scheduled backups?"

> description: "Disable or Mute Alerts during maintenance windows"

Reviewer: We'll need to add a preview image for this, too. Just a line at line five that has:

image: [image url]

Author: I will put something up..

tags: [netdata,alerts,silence,disable,mute,maintenance window]
keywords: [netdata,alerts,silence,disable,mute,maintenance window]
authors: satya
---

The health management APIs in Netdata allows teams to eliminate unnecessary alerting during scheduled maintenance, testing, auto scaling events, and instance reboots.

<!--truncate-->

For all SREs, it is absolutely crucial to filter out expected events during maintenance windows and quickly pinpoint critical issues in your infrastructure. Every minute is crucial while dealing with troubleshooting issues and any distractions that may hijack the troubleshooting process should be subdued.
The health management APIs in Netdata allows teams to eliminate unnecessary alerting during scheduled maintenance, testing, auto scaling events, and instance reboots.
 
## Health Management API

> Netdata provides a command API to control health checks and notifications at runtime. The feature is especially useful for maintenance periods, during which you receive meaningless alerts. The configurations controlled via the API commands is [persisted across Netdata restarts](https://learn.netdata.cloud/docs/agent/web/api/health#persistence).

Reviewer 3: ```suggestion
Netdata provides a command API to control health checks and notifications at runtime. The feature is especially useful for maintenance periods, during which you receive meaningless alerts. The configurations controlled via the API commands are [persisted across Netdata restarts](https://learn.netdata.cloud/docs/agent/web/api/health#persistence).
```

Specifically, the API allows you to:
- Disable health checks completely. Alarm conditions will not be evaluated at all and no entries will be added to the alarm log.
- Silence alarm notifications. Alarm conditions will be evaluated, the alarms will appear in the log and the Netdata UI will show the alarms as active, but no notifications will be sent.
- Disable or Silence specific alarms that match selectors on alarm/template name, chart, context, host and family.

The API is available by default, but it is protected by an api authorization token that is stored in the file you will see in the following entry of http://NODE:19999/netdata.conf:
```yaml
[registry]
   # netdata management api key file = /var/lib/netdata/netdata.api.key
```
 
You can access the API via GET requests, by adding the bearer token to an Authorization http header, like this:
```bash
curl "http://NODE:19999/api/v1/manage/health?cmd=RESET" -H "X-Auth-Token: Mytoken"
```

> By default access to the health management API is only allowed from localhost. Accessing the API from anything else will return a 403 error with the message You are not allowed to access this resource. You can change permissions by editing the allow management from the variable in netdata.conf within the [web server access lists](https://learn.netdata.cloud/docs/agent/web/server#access-lists) section. See web server access lists for more information.

Reviewer 3: ```suggestion
By default, access to the health management API is restricted to localhost. Accessing the API from anywhere else will return a 403 error with the message "You are not allowed to access this resource". You can change permissions by editing the `allow management from` variable in netdata.conf within the [web server access lists](https://learn.netdata.cloud/docs/agent/web/server#access-lists) section. See web server access lists for more information.
```

The command RESET just returns Netdata to the default operation, with all health checks and notifications enabled. If you've configured and entered your token correctly, you should see the plain text response All health checks and notifications are enabled.

**Note**: The Health Management API currently needs to be executed on every host / agent which is under maintenance and at some point in the future we will have a way of triggering maintenance modes from the **Netdata Cloud UI**.

## Disable or silence all alarms

If all you need is temporarily disable all health checks, then you issue the following before your maintenance period starts:
```bash
curl "http://NODE:19999/api/v1/manage/health?cmd=DISABLE ALL" -H "X-Auth-Token: Mytoken"
```
 
The effect of disabling health checks is that the alarm criteria are not evaluated at all and nothing is written in the alarm log. If you want the health checks to be running but to not receive any notifications during your maintenance period, you can instead use this:
```bash
curl "http://NODE:19999/api/v1/manage/health?cmd=SILENCE ALL" -H "X-Auth-Token: Mytoken"
```

> Alarms may then still be raised and logged in Netdata, so you'll be able to see them via the UI.

Reviewer 3: Should we also talk about how the user can just turn off notifications from the UI? They can do this when a maintenance window starts and turn them back on after its done. 

Would be good to drop a screenshot of this as well - the new one that Alexandros demo'd today.

Author: Yes, I think I will add this. Although, the users will currently need to do both - from the Agent and from the Cloud.

Author: Done, but with the old snapshots still.. I will take a look at what Alexandros delivers (if not already in).

- hosts : The hostnames that will need to match.
- families : The alarm families.

You can add any of the selection criteria you need on the request, to ensure that only the alarms you are interested in are matched and disabled/silenced. e.g. there is no reason to add hosts: *, if you want the criteria to be applied to alarms for all hosts.

(*Example 1: Disable all health checks for context = random*)

http://NODE:19999/api/v1/manage/health?cmd=DISABLE&context=random

(*Example 2: Silence all alarms and templates with name starting with out_of on host myhost*)

http://NODE:19999/api/v1/manage/health?cmd=SILENCE&alarm=out_of*&hosts=myhost

(*Example 2.2: Add one more selector, to also silence alarms for cpu1 and cpu2*)

http://NODE:19999/api/v1/manage/health?families=cpu1 cpu2

## List silencers

The command LIST was added in Netdata v1.16.0 and returns a JSON with the current status of the silencers.
```bash
curl "http://NODE:19999/api/v1/manage/health?cmd=LIST" -H "X-Auth-Token: Mytoken"
```

As an example, the following response shows that we have two silencers configured, one for an alarm called samplealarm and one for alarms with context random on host myhost

> ```yaml

Reviewer 4: ```suggestion
```json
```

json
{
       "all": false,
       "type": "SILENCE",
       "silencers": [
               {
                       "alarm": "samplealarm"
               },
               {
                       "context": "random",
                       "hosts": "myhost"
               }
       ]
}
```

The response below shows that we have disabled all health checks.

> ```yaml

Reviewer 4: ```suggestion
```json
```

> json

Reviewer 4: ```suggestion
```

(*Example 2: Silence all alarms and templates with name starting with out_of on host myhost*)

http://NODE:19999/api/v1/manage/health?cmd=SILENCE&alarm=out_of*&hosts=myhost

(*Example 2.2: Add one more selector, to also silence alarms for cpu1 and cpu2*)

http://NODE:19999/api/v1/manage/health?families=cpu1 cpu2

## List silencers

The command LIST was added in Netdata v1.16.0 and returns a JSON with the current status of the silencers.
```bash
curl "http://NODE:19999/api/v1/manage/health?cmd=LIST" -H "X-Auth-Token: Mytoken"
```

As an example, the following response shows that we have two silencers configured, one for an alarm called samplealarm and one for alarms with context random on host myhost

```yaml
json
{
       "all": false,
       "type": "SILENCE",
       "silencers": [
               {
                       "alarm": "samplealarm"
               },
               {
                       "context": "random",
                       "hosts": "myhost"
               }
       ]
}
```

The response below shows that we have disabled all health checks.

```yaml
json
{
       "all": true,
       "type": "DISABLE",
       "silencers": []
}
```
 
 

> **Note**: The system catalog table pg_attribute is usually the largest catalog table. If the pg_attribute table is significantly bloated, a VACUUM FULL operation on the table might require a significant amount of time and might need to be performed separately. The presence of both of these conditions indicate a significantly bloated pg_attribute table that might require a long VACUUM FULL time.

Reviewer 3: This **Note** is from another blog I guess - should be removed.
